adafruit_pybadger/pybadger_base.py

```diff
@@ -92,51 +94,26 @@ class PyBadger:
     BUTTON_A = const(2)
     BUTTON_B = const(1)
 
-    def __init__(self, i2c=None, *, pixels_brightness=1.0):
-        # Accelerometer
-        if i2c is None:
-            try:
-                i2c = board.I2C()
-            except RuntimeError:
-                self._accelerometer = None
-
-        if i2c is not None:
-            int1 = digitalio.DigitalInOut(board.ACCELEROMETER_INTERRUPT)
-            try:
-                self._accelerometer = adafruit_lis3dh.LIS3DH_I2C(i2c, address=0x19, int1=int1)
-            except ValueError:
-                self._accelerometer = adafruit_lis3dh.LIS3DH_I2C(i2c, int1=int1)
-
-        # Buttons
-        self._buttons = GamePadShift(digitalio.DigitalInOut(board.BUTTON_CLOCK),
-                                     digitalio.DigitalInOut(board.BUTTON_OUT),
-                                     digitalio.DigitalInOut(board.BUTTON_LATCH))
+    def __init__(self, *, pixels_brightness=1.0):
+        self._light_sensor = None
+        self._accelerometer = None
 
         # Display
         self.display = board.DISPLAY
         self._display_brightness = 1.0
 
-        # Light sensor
-        self._light_sensor = analogio.AnalogIn(board.A7)
-
-        # PyGamer joystick
-        if hasattr(board, "JOYSTICK_X"):
-            self._pygamer_joystick_x = analogio.AnalogIn(board.JOYSTICK_X)
-            self._pygamer_joystick_y = analogio.AnalogIn(board.JOYSTICK_Y)
-
         # NeoPixels
-        # Count is hardcoded - should be based on board ID, currently no board info for PyBadge LC
-        neopixel_count = 5
-        self._neopixels = neopixel.NeoPixel(board.NEOPIXEL, neopixel_count,
+        self._neopixels = neopixel.NeoPixel(board.NEOPIXEL, self._neopixel_count,
                                             brightness=pixels_brightness, pixel_order=neopixel.GRB)
 
         # Auto dim display based on movement
         self._last_accelerometer = None
         self._start_time = time.monotonic()
 
         # Define audio:
-        self._speaker_enable = digitalio.DigitalInOut(board.SPEAKER_ENABLE)
-        self._speaker_enable.switch_to_output(value=False)
+        if hasattr(board, "SPEAKER_ENABLE"):
```

Comment:
This is checked in a few spots, and then the value of .enable is set to True at the start of several methods, and False at the end. 

It might look a little nicer to set `self._speaker_enable` to None if it's not present, and or/or create a small helper method that does the check.

for example:
```python
if hasattr(board, "SPEAKER_ENABLE"):
  self._speaker_enable = digitalio.DigitalInOut(board.SPEAKER_ENABLE)
else:
  self._speaker_enable = None
```

```python
    def _enable_speaker(self, value=True):
        # or if not hasattr(board, "SPEAKER_ENABLE"):
        if not self._speaker_enable:
            return
        self._speaker_enable.value = value
```

then replace
```python
if hasattr(board, "SPEAKER_ENABLE"):
            self._speaker_enable.value = True
```
with
```python
self._enable_speaker()
```

What do you think?

Reply:
Setting `speaker_enable` to None results in `None type object has no value`. I tried that in the first iteration of getting PyBadger to work on CLUE.

Your method idea ends up saving a few lines of code. I'll incorporate it.